server/api.js

```diff
@@ -816,4 +817,42 @@ router.post("/logout", async (req, res) => {
 	}
 });
 
+router.get("/bid-detail/:id", async (req, res) => {
+	const bidId = req.params.id;
+
+	try {
+		const result = await db.query(
+			`SELECT 
+          b.bid_id, 
+          b.tender_id, 
+          b.bidding_amount, 
+          b.status, 
+          b.bidding_date AS submission_date, 
+          b.suggested_duration_days, 
+          t.title, 
+          t.announcement_date, 
+          t.closing_date, 
+          b.cover_letter, 
+          t.status AS tender_status, 
+          t.description
+       FROM 
+          bid b
+       JOIN 
+          tender t ON b.tender_id = t.id
+       WHERE 
+          b.bid_id = $1
+      `,
+			[bidId]
+		);
+
+		if (result.rows.length === 0) {
+			return res.status(404).json({ code: "BID_NOT_FOUND" });
+		}
+
+		res.status(200).json(result.rows[0]);
```

Comment:
Only the owner of the tender and the owner of the bid should be able to see the details for a bid.

Reply:
i have added permissions in alowlist for this endpoint.
`"/bid-detail": "token",`

Reply:
That's not enough. Not all bidder should be able to see all bids. They should only be able to see the bid if they own it.

The same for buyers. Only buyers that own the tender should be able to see bids for that tender.